arithmetization/src/test/java/net/consensys/linea/zktracer/module/mxp/MxpTest.java

```diff
@@ -335,8 +337,11 @@ private void triggerNonTrivialOrNoop(BytecodeCompiler program, boolean isHalting
       offset1 = EWord.of(util.getRandomBigIntegerByBytesSize(0, MAX_BYTE_SIZE));
       offset2 = EWord.of(util.getRandomBigIntegerByBytesSize(0, MAX_BYTE_SIZE));
 
-      // NOOP case (except for Type2 and Type3 instructions)
-      if (mxpType != MxpType.TYPE_2 && mxpType != MxpType.TYPE_3) {
+      // NOOP case (except for Type2 and Type3 instructions and halting instructions,
+      // as we do not want halting instructions to turn into NOOPs)
```

Comment:
I don't see why we wouldn't want NOOP's triggered by halting instructions. I would imagine we want to test all scenarios, this being one of them. Maybe in a separate test, but it's still of interest to have all ways of triggering a NOOP.

Reply:
Ok.